README
App::colourhexdump



INSTALLATION

This is a Perl module distribution. It should be installed with whichever
tool you use to manage your installation of Perl, e.g. any of

  cpanm .
  cpan  .
  cpanp -i .

Consult http://www.cpan.org/modules/INSTALL.html for further instruction.
Should you wish to install this module manually, the procedure is

  perl Makefile.PL
  make
  make test
  make install

COPYRIGHT AND LICENSE

This software is copyright (c) 2015 by Kent Fredric <kentnl@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


Changes
Revision history for App-colourhexdump

1.000004 2017-03-07T09:37:39Z 5297ae4

1.000003 2017-03-07T09:36:52Z 76602c8
 - Bugfix: Avoid test failures with -Ddefault_inc_excludes_dot

 [Dependencies::Stats]
 - Dependencies changed since 1.000002, see misc/*.deps* for details
 - develop: +1 ↑1 (suggests: ↑2)
 - test: ↓1

1.000002 2015-08-18T05:37:10Z 45f0c28
 [Dependencies::Stats]
 - Dependencies changed since 1.000001, see misc/*.deps* for details
 - configure: +1 (recommends: ↑1)
 - develop: +11 ↑2 -3 (recommends: +1, suggests: ↑2)
 - runtime: -1 (recommends: +1)
 - test: +1 (recommends: ↑1 ↓1)

 [Minor]
 - code shuffled to reduce whitespace
 - dependency on List::MoreUtils removed.

1.000001 2014-10-19T20:42:11Z 7cfeaa2
 [Bugfix]
 - Hyperlinks to preview images were missing due to path migration and gitsubmodule weirdness.
 - This closes #2 ( Thanks Ivan Baidakou for reporting )

1.000000 2014-10-13T23:42:53Z 7b40123
 [00 Major - Version Scheme Change]
 - Migrating from x.yyyzzzab to x.yyyzzz
 - This may affect hand-operated downstreams.
 - https://questhub.io/realm/perl/quest/5388d3a9bbd0be2224000129

 [01 Major - Installer Toolkit Change]
 - Migrating from MB to EUMM
 - This may affect hand-operated downstreams.
 - Critical toolkit and test dependencies only have latest versions advertised as "recommended" instead of required.
 - https://questhub.io/realm/perl/quest/5388cc12bbd0be5b2500011f

 [Dependencies::Stats]
 - Dependencies changed since 0.01011318, see misc/*.deps* for details
 - build: -1
 - configure: +1 -1 (recommends: +1)
 - develop: +48 -2 (recommends: -1, suggests: +1 ↑1)
 - runtime: +1 -1
 - test: +2 ↓1 -2 (recommends: +3)

 [Internals]
 - Replace Class::Load with Module::Runtime

0.01011318 2013-05-21T16:46:25Z
 [Dependencies::Stats]
 - Dependencies changed since 0.01011317, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +1 (suggests: ↑1)

 [Documentation]
 - use Github pages for image instead of github raw access, to be nicer to githubs load.
 - https://github.com/blog/1482-
 - https://help.github.com/articles/using-submodules-with-pages

0.01011317 2013-04-09T21:37:30Z
 - Maintenance Release for Module::Build 0.4004

 [Dependencies::Stats]
 - Dependencies changed since 0.01011316, see misc/*.deps* for details
 - build: ↑1
 - configure: ↑1
 - develop: +5 (recommends: ↑1, suggests: ↑1)
 - test: ↑1

 [Documentation]
 - update copyright year
 - add README.mkdn

 [Meta]
 - bug tracker to github issues

 [Packaging]
 - utilise test_requires in Build.PL

0.01011316 2012-03-12T11:37:10Z
 - Maintenance release.

 [Dependencies::Stats]
 - Dependencies changed since 0.01011315, see misc/*.deps* for details
 - develop: (suggests: ↑1)
 - runtime: +3
 - test: -1

 [Internals]
 - VERSION declaration moved out of BEGIN
 - All packages declare $AUTHORITY

 [Packaging]
 - Move to git versions.
 - Update LICENSE ( Year, Indent, Address )
 - Moved author/release tests to xt/
 - Github uri's moved to https://
 - Declare x_authority

 [Tests]
 - regenerated 00-compile.t

0.01011315 2011-04-09T07:44:09Z
 [Bugfix]
 - rt#67341 : Moose bailing due to role collision should be solved in this release. It would appear I was implicitly
   importing the same role twice for some reason. Thanks Andreas J. König ( ANDK ) for reporting.

 [Dependencies::Stats]
 - Dependencies changed since 0.01000103, see misc/*.deps* for details
 - develop: +1 -1 (recommends: +1 -1, suggests: +1 -1)
 - runtime: ↑1 -1

 [Internal Notes]
 - previously used "with qw( ::GLD ::Dashes )" in main package, as if memory  serves correctly, at some past point this
   was required for correct behaviour. However, this is no longer true, and ::Dashes now imports ::GLD, so if they're
   both  specified by us, it will be like importing ::GLD twice, which will obviously trigger a symbol collision.

 [Packaging]
 - moved to @Author::KENTNL
 - Normalise Changelog for CPAN::Changes.
 - ship perltidyrc

 [Tests]
 - Adds CPAN::Changes tests
 - Drops portability tests

0.01000103 2010-12-20T19:02:48Z
 - Depend on Term::ANSIColour 3.00. A Few silly people don't have that.
 - Add a doclink to the image

 [Dependencies::Stats]
 - Dependencies changed since 0.01000020, see misc/*.deps* for details
 - runtime: ↑1

0.01000020 2010-12-20T12:42:25Z
 - First version.


Makefile.PL
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v6.009.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "HexDump, but with character-class highlighting.",
  "AUTHOR" => "Kent Fredric <kentnl\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "App-colourhexdump",
  "EXE_FILES" => [
    "bin/colorhexdump",
    "bin/colourhexdump"
  ],
  "LICENSE" => "perl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "App::colourhexdump",
  "PREREQ_PM" => {
    "Carp" => 0,
    "Getopt::Long::Descriptive" => 0,
    "Module::Runtime" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "MooseX::Getopt::Dashes" => "0.37",
    "String::RewritePrefix" => 0,
    "Term::ANSIColor" => "3.00",
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => 0
  },
  "VERSION" => "1.000004",
  "test" => {
    "TESTS" => "t/*.t t/00-compile/*.t"
  }
);


my %FallbackPrereqs = (
  "Carp" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Getopt::Long::Descriptive" => 0,
  "Module::Runtime" => 0,
  "Moose" => 0,
  "Moose::Role" => 0,
  "MooseX::Getopt::Dashes" => "0.37",
  "String::RewritePrefix" => 0,
  "Term::ANSIColor" => "3.00",
  "Test::More" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);


lib/App/colourhexdump.pm
use 5.006;    # our
use strict;
use warnings;

package App::colourhexdump;

our $VERSION = '1.000004';

# ABSTRACT: HexDump, but with character-class highlighting.

our $AUTHORITY = 'cpan:KENTNL'; # AUTHORITY

use Moose qw( has with );
use MooseX::Getopt::Dashes 0.37;
with qw( MooseX::Getopt::Dashes );

use Getopt::Long::Descriptive;
use Term::ANSIColor 3.00 qw( colorstrip );
use App::colourhexdump::Formatter;
use namespace::autoclean;

has colour_profile => (
  metaclass     => 'Getopt',
  isa           => 'Str',
  is            => 'rw',
  default       => 'DefaultColourProfile',
  cmd_aliases   => [qw/ C color-profile /],
  documentation => 'Backend to use for colour highlighting (DefaultColourProfile)',
);

has row_length => (
  metaclass     => 'Getopt',
  isa           => 'Int',
  is            => 'ro',
  default       => 32,
  cmd_aliases   => [qw/ r row /],
  documentation => 'Number of bytes per display row (32).',

);

has chunk_length => (
  metaclass     => 'Getopt',
  isa           => 'Int',
  is            => 'rw',
  default       => 4,
  cmd_aliases   => [qw/ x chunk /],
  documentation => 'Number of bytes per display hex display group (4).',
);

has _files => (
  metaclass     => 'Getopt',
  isa           => 'ArrayRef[Str]',
  is            => 'rw',
  default       => sub { [] },
  cmd_flag      => 'file',
  cmd_aliases   => [qw/ f /],
  documentation => 'Add a file to the list of files to process. \'-\' for STDIN.',

);

has 'show_file_prefix' => (
  metaclass     => 'Getopt',
  isa           => 'Bool',
  is            => 'rw',
  default       => 0,
  documentation => 'Enable printing the filename on the start of every line ( off ).',

);
has 'show_file_heading' => (
  metaclass     => 'Getopt',
  isa           => 'Bool',
  is            => 'rw',
  default       => 0,
  documentation => 'Enable printing the filename before the hexdump output. ( off ).',
);
has 'colour' => (
  metaclass     => 'Getopt',
  isa           => 'Bool',
  is            => 'rw',
  default       => 1,
  cmd_aliases   => [qw/ c color /],
  documentation => 'Enable coloured output ( on ). --no-colour to disable.',

);

__PACKAGE__->meta->make_immutable;
no Moose;









sub BUILD {
  my $self = shift;
  push @{ $self->_files() }, @{ $self->extra_argv };
  return $self;

}









sub get_filehandle {
  my ( undef, $filename ) = @_;
  if ( q[-] eq $filename ) {
    return \*STDIN;
  }
  require Carp;
  open my $fh, '<', $filename or Carp::confess("Cant open $_ , $!");
  return $fh;
}









sub run {
  my $self = shift;
  if ( not @{ $self->_files } ) {
    push @{ $self->_files }, q[-];
  }
  ## no critic ( Variables::RequireLocalizedPunctuationVars )
  local $ENV{ANSI_COLORS_DISABLED} = $ENV{ANSI_COLORS_DISABLED};
  if ( not $self->colour ) {
    $ENV{ANSI_COLORS_DISABLED} = 1;
  }
  for ( @{ $self->_files } ) {
    my $prefix = q{};
    if ( $self->show_file_prefix ) {
      $prefix = $_;
    }
    if ( length $prefix ) {
      $prefix .= q{:};
    }
    ## no critic ( RequireCheckedSyscalls );
    if ( $self->show_file_heading ) {
      print qq{- Contents of $_ --\n};
    }
    my $formatter = App::colourhexdump::Formatter->new(
      colour_profile => $self->colour_profile,
      row_length     => $self->row_length,
      chunk_length   => $self->chunk_length,
    );

    $formatter->format_foreach_in_fh(
      $self->get_filehandle($_),
      sub {
        print $prefix . shift;
      },
    );
  }
  return 1;
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

App::colourhexdump - HexDump, but with character-class highlighting.

=head1 VERSION

version 1.000004

=head1 SYNOPSIS

    usage: colourhexdump [-?Ccfrx] [long options...]
        -? --usage --help                     Prints this usage information.
        --color-profile -C --colour-profile   Backend to use for colour highlighting (DefaultColourProfile)
        --row -r --row-length                 Number of bytes per display row (32).
        --chunk -x --chunk-length             Number of bytes per display hex display group (4).
        -f --file                             Add a file to the list of files to process. '-' for STDIN.
        --show-file-prefix                    Enable printing the filename on the start of every line ( off ).
        --show-file-heading                   Enable printing the filename before the hexdump output. ( off ).
        --color -c --colour                   Enable coloured output ( on ). --no-colour to disable.

It can be used like so

    colourhexdump  file/a.txt file/b.txt -- --this-is-treated-like-a-file.txt

If you are using an HTML-enabled POD viewer, you should see a screenshot of this in action:

( Everyone else can visit L<http://kentnl.github.io/App-colourhexdump/media/Screenshot.png> )

=for html <center><img src="http://kentnl.github.io/App-colourhexdump/media/Screenshot.png" alt="Screenshot with explanation of colours" width="826" height="838"/></center>

=head1 METHODS

=head2 BUILD

This just pushes extra_argv from getopt into the files list.

B<INTERNAL>

=head2 get_filehandle

    my $fh = $self->get_filehandle( $filename_or_stdindash );

B<INTERNAL>

=head2 run

Run the app.

    App::colourhexdump->new_with_options()->run();

=head1 AUTHOR

Kent Fredric <kentnl@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Kent Fredric <kentnl@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/App/colourhexdump/Formatter.pm
use 5.006;    # our
use strict;
use warnings;

package App::colourhexdump::Formatter;

our $VERSION = '1.000004';

# ABSTRACT: Colour-Highlight lines of data as hex.

our $AUTHORITY = 'cpan:KENTNL'; # AUTHORITY

use Moose qw( has );
use String::RewritePrefix;
use Module::Runtime qw( require_module );
use Term::ANSIColor 3.00 qw( colorstrip );

use namespace::autoclean;

has colour_profile => (
  does       => 'App::colourhexdump::ColourProfile',
  is         => 'rw',
  lazy_build => 1,
  init_arg   => undef,
);

has real_colour_profile_class => (
  isa        => 'Str',
  is         => 'rw',
  lazy_build => 1,
  init_arg   => undef,
);

has colour_profile_class => (
  isa      => 'Str',
  is       => 'rw',
  init_arg => 'colour_profile',
  default  => 'DefaultColourProfile',
);

has row_length => (
  isa     => 'Int',
  is      => 'ro',
  default => 32,
);

has chunk_length => (
  isa     => 'Int',
  is      => 'rw',
  default => 4,
);

has hex_row_length => (
  isa        => 'Int',
  is         => 'rw',
  lazy_build => 1,
  init_arg   => undef,
);

__PACKAGE__->meta->make_immutable;
no Moose;

sub _build_hex_row_length {
  my $self = shift;

  # Each byte takes 2 bytes to print.
  #
  if ( $self->chunk_length > $self->row_length ) {
    $self->chunk_length( $self->row_length );
  }
  my $real_chunk_length = $self->chunk_length * 2;

  my $chunks     = int( $self->row_length / $self->chunk_length );
  my $extrachunk = 0;

  if ( ( $chunks * $self->chunk_length ) < $self->row_length ) {
    $extrachunk = $self->row_length - ( $chunks * $self->chunk_length );
  }

  my $whitespaces = $chunks - 1;
  if ( $extrachunk > 0 ) {
    $whitespaces++;
  }

  return ( $chunks * $real_chunk_length ) + $whitespaces + $extrachunk;

}










## no critic ( Subroutines::RequireArgUnpacking )

sub format_foreach_in_fh {
  my ( $self, $fh, $callback ) = ( $_[0], $_[1], $_[2] );
  my $offset = 0;
  while ( read $fh, my $buffer, $self->row_length ) {
    $callback->( $self->format_row( $buffer, $offset ) );
    $offset += $self->row_length;
  }
  return 1;
}







## no critic ( Subroutines::RequireArgUnpacking )

sub format_row_from_fh {
  my ( $self, $fh, $offset ) = ( $_[0], $_[1], $_[2] );
  read $fh, my $buffer, $self->row_length or return;
  my $str = $self->format_row( $buffer, $offset );
  $offset += $self->row_length;
  return $str, $offset;
}







sub format_row {
  my ( $self, $row, $offset ) = @_;

  my $format = "%10s: %s   %s\n";
  my $offset_hex = _to_hex( pack q{N*}, $offset );

  my @chars = split //, $row;

  return sprintf $format, $offset_hex, $self->pad_hex_row( $self->hex_encode(@chars) ), $self->pretty_encode(@chars);
}







sub hex_encode {
  my ( $self, @chars ) = @_;
  my @out;
  while ( my @vals = splice @chars, 0, $self->chunk_length, () ) {
    my $chunk;
    for (@vals) {
      $chunk .= $self->colour_profile->get_string_pre($_);
      $chunk .= _to_hex($_);
      $chunk .= $self->colour_profile->get_string_post($_);
    }
    push @out, $chunk;
  }
  return join q{ }, @out;
}







sub pretty_encode {
  my ( $self, @chars ) = @_;
  my $output;
  for (@chars) {
    $output .= $self->colour_profile->get_string_pre($_);
    $output .= $self->colour_profile->get_display_symbol_for($_);
    $output .= $self->colour_profile->get_string_post($_);
  }
  return $output;
}

sub _to_hex {
  return join q{}, map { unpack q{H*}, $_ } @_;
}







sub pad_hex_row {
  my ( $self, $row ) = @_;
  my $length = length colorstrip($row);
  if ( $length > $self->hex_row_length ) {
    return $row;
  }
  return $row . ( q{ } x ( $self->hex_row_length - $length ) );
}

sub _build_colour_profile {
  my $self = shift;
  require_module( $self->real_colour_profile_class );
  return $self->real_colour_profile_class->new();
}

sub _build_real_colour_profile_class {
  my $self = shift;
  return String::RewritePrefix->rewrite( { q{} => 'App::colourhexdump::', q{=} => q{} }, $self->colour_profile_class );
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

App::colourhexdump::Formatter - Colour-Highlight lines of data as hex.

=head1 VERSION

version 1.000004

=head1 METHODS

=head2 format_foreach_in_fh

    $formatter->format_foreach_in_fh( $fh, sub {
        my $formatted = shift;
        print $formatted;
    });

=head2 format_row_from_fh

    my ( $formatted , $offset ) = $formatter->format_row_from_fh( $fh, $offset );

=head2 format_row

    my $formatted = $formatter->format_row( "Some Characters", $offset );

=head2 hex_encode

    my $hexes = $formatter->hex_encode( split //, "Some Characters" );

=head2 pretty_encode

    my $nicetext = $formatter->pretty_encode( split //, "Some Characters" );

=head2 pad_hex_row

    my $padded = $Formatter->pad_hex_row( $formatter->hex_enode( split //, "Some Characters" ) );

=head1 AUTHOR

Kent Fredric <kentnl@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Kent Fredric <kentnl@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/App/colourhexdump/ColourProfile.pm
use 5.006;    # our
use strict;
use warnings;

package App::colourhexdump::ColourProfile;

our $VERSION = '1.000004';

# ABSTRACT: A Role for Colour Profiles

our $AUTHORITY = 'cpan:KENTNL'; # AUTHORITY

use Moose::Role qw( requires );
use namespace::autoclean;















requires 'get_colour_for';









requires 'get_display_symbol_for';

no Moose::Role;

use Term::ANSIColor 3.00 qw( RESET );







## no critic ( RequireArgUnpacking )

sub get_string_pre {
  my ( $self, $char ) = ( $_[0], $_[1] );
  my $colourcode = $self->get_colour_for($char);
  if ( defined $colourcode ) {
    return $colourcode;
  }
  return q{};
}








## no critic ( RequireArgUnpacking )

sub get_string_post {
  my ( $self, $char ) = ( $_[0], $_[1] );
  my $colourcode = $self->get_colour_for($char);
  if ( defined $colourcode ) {
    return RESET;
  }
  return q{};
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

App::colourhexdump::ColourProfile - A Role for Colour Profiles

=head1 VERSION

version 1.000004

=head1 SYNOPSIS

    package App::colourhexdump::ColourProfileName

    use Moose;
    with qw( App::colourhexdump::ColourProfile );

    sub get_colour_for {
        my ( $self, $char ) = @_ ;
        ...
        return "\e[31m" if /badthings/;
        return undef;    # don't colour
    }
    sub get_display_symbol_for {
        my ($self, $char) = @_ ;
        ...
        return '.' if $char =~ /badthings/
        return $char;        # printable
    }

=head1 REQUIRED METHODS

=head2 C<get_colour_for>

    my $colour = $object->get_colour_for( "\n" );

Return any string of data that should be prepended every time a given character is seen.

Generally, you only want to print ANSI Escape codes.

Don't worry about resetting things, we put a C<^[[0m> in for you.

Return C<undef> if you do not wish to apply colouring.

=head2 C<get_display_symbol_for>

    my $symbol = $object->get_display_symbol_for( "\n" );

Returns a user viewable alternative to the matched string.

=head1 METHODS

=head2 C<get_string_pre>

Wraps L</get_colour_for> and returns either a string sequence or ''.

=head2 C<get_string_post>

Wraps L</get_colour_for> and returns either an ANSI Reset Code, or '', depending
on what was returned.

=head1 AUTHOR

Kent Fredric <kentnl@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Kent Fredric <kentnl@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


lib/App/colourhexdump/DefaultColourProfile.pm
use 5.006;
use strict;
use warnings;

package App::colourhexdump::DefaultColourProfile;

our $VERSION = '1.000004';

# ABSTRACT: The default colour profile

our $AUTHORITY = 'cpan:KENTNL'; # AUTHORITY

use Moose qw( has with );
use namespace::autoclean;

with 'App::colourhexdump::ColourProfile';

__PACKAGE__->meta->make_immutable;
no Moose;

use Term::ANSIColor 3.00 qw( BLACK ON_RED BOLD BRIGHT_BLUE BRIGHT_YELLOW ON_YELLOW RED BLUE YELLOW );







## no critic ( Subroutines::RequireArgUnpacking )
sub get_colour_for {
  return BLACK . ON_RED                 if $_[1] =~ /\r/;
  return BOLD . BRIGHT_BLUE             if $_[1] =~ /\n/;
  return BLUE                           if $_[1] =~ / /;
  return BOLD . BRIGHT_BLUE . ON_YELLOW if $_[1] =~ /\t/;
  return RED                            if $_[1] =~ qr{[^[:print:]]};
  return                                if $_[1] =~ qr{[[:alpha:]\d]};
  return YELLOW;
}







## no critic ( Subroutines::RequireArgUnpacking )

sub get_display_symbol_for {
  return q{_} if $_[1] =~ qr{\s};
  return q{.} if $_[1] =~ qr{[^[:print:]]};
  return $_[1];
}

1;

__END__

=pod

=encoding UTF-8

=head1 NAME

App::colourhexdump::DefaultColourProfile - The default colour profile

=head1 VERSION

version 1.000004

=head1 SYNOPSIS

This is the default colour profile.

    \r           => black on red               '_'
    \n           => bold bright blue           '_'
    " "          => blue                       '_'
    \t           => bold bright blue on yellow '_'
    alphanumeric => white                      $char
    nonprintable => red                        '.'
    everything else => yellow                  $char

Invocation:

    my $cp = App::colourhexdump::DefaultColourProfile->new();
    my $colouredChar =
        $cp->get_string_pre( $char ) .
        $cp->get_dispaly_symbol_for( $char ) .
        $cp->get_string_post( $char );

=head1 METHODS

=head2 get_colour_for

See L<App::colourhexdump::ColourProfile/get_colour_for>

=head2 get_display_symbol_for

See L<App::colourhexdump::ColourProfile/get_display_symbol_for>

=head1 AUTHOR

Kent Fredric <kentnl@cpan.org>

=head1 COPYRIGHT AND LICENSE

This software is copyright (c) 2017 by Kent Fredric <kentnl@cpan.org>.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.

=cut


Build.PL

# This file was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
use strict;
use warnings;

use Module::Build 0.28;


my %module_build_args = (
  "build_requires" => {
    "Module::Build" => "0.4210"
  },
  "configure_requires" => {
    "Module::Build" => "0.4210"
  },
  "dist_abstract" => "HexDump, but with character-class highlighting.",
  "dist_author" => [
    "Kent Fredric <kentnl\@cpan.org>"
  ],
  "dist_name" => "App-colourhexdump",
  "dist_version" => "1.000000",
  "license" => "perl",
  "module_name" => "App::colourhexdump",
  "recommends" => {},
  "recursive_test_files" => 1,
  "requires" => {
    "Carp" => 0,
    "Class::Load" => "0.06",
    "Getopt::Long::Descriptive" => 0,
    "List::MoreUtils" => 0,
    "Moose" => 0,
    "Moose::Role" => 0,
    "MooseX::Getopt::Dashes" => "0.37",
    "String::RewritePrefix" => 0,
    "Term::ANSIColor" => "3.00",
    "namespace::autoclean" => 0,
    "perl" => "5.006",
    "strict" => 0,
    "warnings" => 0
  },
  "script_files" => [
    "bin/colorhexdump",
    "bin/colourhexdump"
  ],
  "test_requires" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "Test::More" => "1.001006"
  }
);


my %fallback_build_requires = (
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "Module::Build" => "0.4210",
  "Test::More" => "1.001006"
);


unless ( eval { Module::Build->VERSION(0.4004) } ) {
  delete $module_build_args{test_requires};
  $module_build_args{build_requires} = \%fallback_build_requires;
}

my $build = Module::Build->new(%module_build_args);

$build->create_build_script;
